src/main/java/org/apache/commons/imaging/formats/dcx/DcxImageParser.java

```diff
@@ -172,7 +173,7 @@ private DcxHeader readDcxHeader(final ByteSource byteSource) throws ImagingExcep
     public void writeImage(final BufferedImage src, final OutputStream os, final PcxImagingParameters params) throws ImagingException, IOException {
         final int headerSize = 4 + 1024 * 4;
 
-        final BinaryOutputStream bos = BinaryOutputStream.littleEndian(os);
+        final BinaryOutputStream bos = BinaryOutputStreamFactory.create(os, ByteOrder.LITTLE_ENDIAN);
```

Comment:
This is much worse IMO. You're trading a clear factory method with 1 arg for another with 2. Worse, the PR allows for null input where none makes sense, and then would throw an exception anyway. The current code does not suffer from this confusion.